includes/REST_API/Stories_Base_Controller.php

```diff
@@ -58,8 +68,151 @@ class Stories_Base_Controller extends WP_REST_Posts_Controller {
 	 */
 	public function __construct( $post_type ) {
 		parent::__construct( $post_type );
-		$this->namespace = 'web-stories/v1';
-		$this->decoder   = new Decoder();
+		$this->namespace   = 'web-stories/v1';
+		$this->decoder     = new Decoder();
+		$this->experiments = new Experiments();
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @see register_rest_route()
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		parent::register_routes();
+
+		if ( ! $this->experiments->is_experiment_enabled( 'enablePostLocking' ) ) {
+			return;
+		}
+
+		$lock_args = [];
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\d]+)/lock',
+			[
+				'args' => [
+					'id' => [
+						'description' => __( 'Unique identifier for the object.', 'web-stories' ),
+						'type'        => 'integer',
+					],
+				],
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'update_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+				[
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => [ $this, 'delete_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success.
+	 */
+	public function get_lock( $request ) {
+		return $this->prepare_lock_for_response( $request );
+	}
+
+	/**
+	 * Update post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success.
+	 */
+	public function update_lock( $request ) {
+		if ( ! function_exists( 'wp_set_post_lock' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/post.php';
+		}
+
+		wp_set_post_lock( $request['id'] );
+
+		return $this->prepare_lock_for_response( $request );
+	}
+
+	/**
+	 * Delete post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response Response object on success.
+	 */
+	public function delete_lock( $request ) {
+		$previous = $this->prepare_lock_for_response( $request );
+		$result   = delete_post_meta( $request['id'], '_edit_lock' );
```

Comment:
I wonder why `wp_ajax_wp_remove_post_lock()` doesn't simply `delete_post_meta` as well but instead uses `update_post_meta` with an updated time.

Does that perhaps save a DB query?

Reply:
I don't know. Willing to change it, but see how it hurts to delete the post meta here.